pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -256,6 +380,23 @@ func TestPlugin(t *testing.T) {
 				},
 			},
 		},
+		"claim-reference-structured": {
+			pod:    podWithClaimName,
+			claims: []*resourcev1alpha2.ResourceClaim{structuredAllocatedClaim, otherClaim},
+			want: want{
+				prebind: result{
+					changes: change{
+						claim: func(claim *resourcev1alpha2.ResourceClaim) *resourcev1alpha2.ResourceClaim {
+							if claim.Name == claimName {
+								claim = claim.DeepCopy()
+								claim.Status.ReservedFor = inUseClaim.Status.ReservedFor
+							}
+							return claim
+						},
+					},
+				},
+			},
+		},
```

Comment:
TBH, I have zero idea what is happening here. There are way too many things hidden in abstractions.

what is podWithClaimName? why makes it structured? what does `change` receive?

https://abseil.io/resources/swe-book/html/ch12.html#:~:text=Tests%20and%20Code%20Sharing%3A%20DAMP%2C%20Not%20DRY

Reply:
Pods are not different, only claims are (potentially different parameter reference, different status). The way how I arrived at some of these unit tests is that I copied existing ones, swapped out the claims for their counterparts and checked what the expected result should be. This gave me roughly coverage of the same scenarios that I had before.

I then thought about some more special cases that can only occur when using structured parameters.

I understand that it's hard to verify that this covers all corner cases. I checked code coverage, but that's only a partial answer.

Regarding "want changes": this is a helper to describe the expected changes more concisely. In this case, the only operation which has any visible effect in the things that are getting checked (cluster objects, assume cache, in-flight claims) is "prebind".

I think it is easier to see what the effect is when describing what changes. Otherwise the test case would have to list the full expected state for each operation and it would be very hard to see that prebind does something.